Source/Core/DiscIO/RiivolutionPatcher.cpp

```diff
@@ -529,8 +530,15 @@ static void ApplyMemoryPatch(u32 offset, const std::vector<u8>& value,
   if (!original.empty() && !MemoryMatchesAt(offset, original))
     return;
 
-  for (u32 i = 0; i < value.size(); ++i)
+  const u32 size = static_cast<u32>(value.size());
+  for (u32 i = 0; i < size; ++i)
     PowerPC::HostTryWriteU8(value[i], offset + i);
+  const u32 overlaps = HLE::UnpatchRange(offset, offset + size);
+  if (overlaps)
```

Comment:
```suggestion
  if (overlaps != 0)
```
You may also want to consider renaming the variable to something like `overlapping_hook_count` to make the check clearer.